### DISCUSSION TITLE
Cache invalidation on code fail?

### DISCUSSION THREAD

**Discussion by FyZzyss** [Krukov/cashews#40] [Q&A] [answered]
I would be glad to see an example of how I can reset the cache if another method of the class gets an error.

**Comment by simon-liebehenschel** [comment, marked as answer]
I never think about such cache use cases. The first thing that comes to my mind is using a decorator that catches exceptions and performs any callbacks you need when an error happens. Simplified example:

```python    
cache_backend = AnyCacheBackendYouWant()


def clear_cache_on_error(function): 
      # I do not write full decorator's body for readability
        
        try:
            return await function(*args, **kwargs)
        except Exception as error:
               # do anything you need...
               await cache_backend.clear()
               raise



class YourClass:
    
    @cached(cache=cache_backend)
    @clear_cache_on_error
    def func(self):
        ...
```

**Reply by FyZzyss** [reply]
Thanks for the quick response. 
I can tell you a little bit about my case. Maybe this will serve as an idea for new features;) 
I will use the cache when integrating with a third-party service to store the token. They have a documented token lifetime, but I don't really trust them) I added a lifetime according to the documentation, but additionally I plan to clear the cache and make a new request with a new token as soon as I get a response code from them that the token has expired/invalid.
